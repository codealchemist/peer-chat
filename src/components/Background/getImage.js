const getImage = () => {
  const min = 0
  const max = 33
  const i = Math.floor(Math.random() * (max - min + 1)) + min
  const image = require(`./backgrounds/background${i}.jpg`)
  return image
}

export default getImage
