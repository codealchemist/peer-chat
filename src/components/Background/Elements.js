import styled from 'styled-components'
import getImage from './getImage'

export const Wrapper = styled.div`
  background: url(${getImage()}) no-repeat;
  background-size: cover;
  background-position: center;
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: -1;
  filter: blur(5px);
`

export const Fade = styled.div`
  width: 100%;
  height: 100%;
  background: black;
  opacity: 0.7;
`
