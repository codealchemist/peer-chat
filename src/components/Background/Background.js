import React from 'react'
import { Wrapper, Fade } from './Elements'

const Background = () => (
  <Wrapper>
    <Fade />
  </Wrapper>
)

export default Background
