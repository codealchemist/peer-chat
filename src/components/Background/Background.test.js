import React from 'react'
import { mount } from 'enzyme'
import Background from './Background'

jest.mock('./Elements', () => ({
  Wrapper: 'div',
  Fade: 'div'
}))

describe('Background', () => {
  it('should render', () => {
    const wrapper = mount(<Background />)
    expect(wrapper).toMatchSnapshot()
  })
})
