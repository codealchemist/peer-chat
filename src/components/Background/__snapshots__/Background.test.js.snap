// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background should render 1`] = `
<Background>
  <div>
    <div />
  </div>
</Background>
`;
